src/commands/project/upgrade.js

```diff
@@ -1,16 +1,58 @@
 // @flow
 import * as options from '../../utils/options';
 import { BoltError } from '../../utils/errors';
+import Project from '../../Project';
+import Package from '../../Package';
+import type { Dependency } from '../../types';
+import upgradeDependenciesInPackage from '../../utils/upgradeDependenciesInPackages';
 
-export type ProjectUpgradeOptions = {};
+// TODO: pass flags individially, upgrade has many flags this is here for testing
+function toScriptFlags(flags: options.Flags) {
+  let scriptFlags = [];
+
+  Object.keys(flags).map(flag => {
+    if (flag === '--') return;
+    if (typeof flags[flag] === 'string') {
+      scriptFlags.push(`--${flag}=${flags[flag]}`);
+    } else {
+      scriptFlags.push(`--${flag}`);
+    }
+  });
+
+  return scriptFlags;
+}
+
+export type ProjectUpgradeOptions = {
+  cwd?: string,
+  deps: Array<Dependency>,
+  flags: Array<string>
+};
 
 export function toProjectUpgradeOptions(
   args: options.Args,
   flags: options.Flags
 ): ProjectUpgradeOptions {
-  return {};
+  let depsArgs = [];
+
+  args.forEach(dep => {
+    depsArgs.push(options.toDependency(dep));
+  });
+
+  return {
+    cwd: options.string(flags.cwd, 'cwd'),
+    deps: depsArgs,
+    flags: toScriptFlags(flags)
+  };
 }
 
 export async function projectUpgrade(opts: ProjectUpgradeOptions) {
-  throw new BoltError('Unimplemented command "project upgrade"');
+  let cwd = opts.cwd || process.cwd();
+  let project = await Project.init(cwd);
+  let pkg = await Package.closest(cwd);
+
+  try {
+    await upgradeDependenciesInPackage(project, pkg, opts.deps, opts.flags);
```

Comment:
I think second argument should be `project.pkg` , can you please very if we run this command in workspaces it upgrades in project ?

Reply:
It works. But I have tested with single level of dependency and project.pkg makes sense here